### ISSUE TITLE
Osc comands do not work 

### ISSUE THREAD

**Issue by frankbash** [magdesign/PocketVJ-CP-v3#69]
CP 2.4.0f the osc commands do nothing on port 9876 (after activation) only the hp-player still works under the old port 9000, any ideas? the pi is connected via lan, touchosc sends to the ip 192.168.2.16 (the pi on the router) and the port 9876

**Comment by magdesign** [comment]
Did you activate OSC in the ControlPanel ?

**Comment by frankbash** [comment]
yes i activated it and in the status window it´s show that activated

**Comment by ProjectileObjects** [comment]
Hi Frank, when I was testing OSC I found it better to run the command directly in terminal for testing.

SSH in and run the node command for osc_control.js   It should say it is running and when you send it commands, the prompt will echo back immediately. more here: [https://github.com/magdesign/PocketVJ-CP-v3/blob/master/Updating%20PocketVJ%20from%20web%20CP%202.md](url)

If you are on mac, download QLab and send OSC commands to the PVJ, very easy to set up.  LMK if you would like to try it that way.

**Comment by frankbash** [comment]
hey i used your touchosc template and want to use an iPad later, maybe mag design fix this in a update, so osc is alway on after boot up the pj,

the work around with ssh and qlab may work for one pj but not if u use more then one

**Comment by magdesign** [comment]
Its not broken, its "off" by default.
If you "Autostart" into "Scheduler/OSC" OSC will be activated everytime you boot up.

**Comment by ProjectileObjects** [comment]
> 
> 
> hey i used your touchosc template and want to use an iPad later, maybe mag design fix this in a update, so osc is alway on after boot up the pj,
> 
> the work around with ssh and qlab may work for one pj but not if u use more then one

Not true. I've used tmux-cssh and Qlab to automate multiple pocketVJ's at once.  It's very easy to do.  Qlab's network settings are where you can specify which devices to send to, then you use Auto-continue triggers and if you start one, it will immediately trigger all of the others.  With terminal open and tmux-cssh showing multiple PVJ in parallel, the command prompt updates as expected.

**Comment by frankbash** [comment]
maybe you should make a video, so i can see this setup

**Comment by ProjectileObjects** [comment]
> maybe you should make a video, so i can see this setup

I don't have time to do that now, but I have an article about it planned a few months into 2019.  For now, I'll share some photos from a tech doc I drafted a while back.

Terminal with tmux-cssh
https://ws1.sinaimg.cn/large/006tNbRwly1fydt8ru0wxj311w0qktea.jpg

Network Settings in QLab
https://ws2.sinaimg.cn/large/006tNbRwly1fydt9dbtp0j31cz0u0dwg.jpg

QLab OSC commands (Note: these are set to Auto-Follow, you will want to use Auto-Continue if you want them to happen at the same time).
https://ws3.sinaimg.cn/large/006tNbRwly1fydt9kdb99j311j0u0gww.jpg

After running the QLab sequence:
https://ws2.sinaimg.cn/large/006tNbRwly1fydt9mzcfkj311w0qkgrd.jpg

Another example to Play a video (/startmaster01), pause it after 5 seconds (/pause), and trigger play again with the space bar (or Go command)  (/pause) <- Same command un-pauses the video. Finally another space bar will stop video video entirely (/stopall)
https://ws4.sinaimg.cn/large/006tNbRwly1fydt9vvb3zj31i40c20w2.jpg

**Comment by frankbash** [comment]
autostart in osc don´t work(blank screen and no osc), manually start osc don´t work, osc with the port 9876 don´t work
ssh to pi work, but directory /osc does not exist 
![bildschirmfoto 2018-12-20 um 23 45 40](https://user-images.githubusercontent.com/6725467/50315249-4cc24280-04b2-11e9-921a-d963e718fb75.png)

qlab settings 
![bildschirmfoto 2018-12-20 um 23 47 25](https://user-images.githubusercontent.com/6725467/50315273-62d00300-04b2-11e9-801b-ee7d2b99cb16.png)
![bildschirmfoto 2018-12-20 um 23 46 33](https://user-images.githubusercontent.com/6725467/50315274-6499c680-04b2-11e9-9e7f-1f6a1004bc8d.png)

only the hplayer osc commandos still work in qlab ( Port 9000) 

any further ideas?

**Comment by ProjectileObjects** [comment]
> /osc does not exist

"then that's your problem"

If /osc doesn't exist then auto start won't do anything.  Did you install node npm and follow the instructions in the readme to transfer the osc_control.js file to the /osc directory?

Found here:
https://github.com/magdesign/PocketVJ-CP-v3/blob/master/Updating%20PocketVJ%20from%20web%20CP%202.md

**Comment by magdesign** [comment]
Which version of PocketVJ do you have?

OSC is implemented since version 3.6
https://github.com/magdesign/PocketVJ-CP-v3/blob/master/RELEASE_NOTES.md

 If you have an older version you need to install all by hand as ProjectileObjects stated.

**Comment by frankbash** [comment]
ah there's the error I'm on version 3.5,
thanks for the help and merry Christmas

**Comment by frankbash** [comment]
<img width="1260" alt="bildschirmfoto 2018-12-28 um 11 40 52" src="https://user-images.githubusercontent.com/6725467/50513056-aec40e80-0a95-11e9-964e-9c27ec098855.png">
nodejs network is not reachable :( same on all terminals , so its not very convenient to update the pocketvj :( so my bought sd-card (on 14.07.2018 ver.3.5) is now old after 5 month :( so what are your plans in future, how can customer update the pocketvj the easy way?????

**Comment by magdesign** [comment]
You have to connect your PocketVJ to the internet or download the file on another machine and manually copy and install it to the PocketVJ. 

For the plans, see the roadmap: https://github.com/magdesign/PocketVJ-CP-v3/blob/master/ROADMAP.md

**Comment by frankbash** [comment]
the pi is hooked up internet via rj45, manually download is done so where to put the file?????
this need a tutorial, how to upgrade from version 3.5 to the lastest firmware and not only how to update the control panel,  i expected more support for 149chf

**Comment by magdesign** [comment]
Did you read this: https://github.com/magdesign/PocketVJ-CP-v3/blob/master/Updating%20PocketVJ%20from%20web%20CP%202.md

Here is everything described on how to install OSC.
You can copy the downloaded file anywhere and extract it. Do you have skills on terminal?
What more support did you expected then?

**Comment by frankbash** [comment]
finale i made it work (with my terminal basic skills and an other tutorial how to instal nodejs) osc is working cp is 2.4.1 and pvj is 3.5

and now something killed the control panel, chrome shows everything like before but when i press a button nothing happen, the tabs and file explorer still working fine, but the rest is dead

any ideas?

**Comment by magdesign** [comment]
Great to hear that you could install node :-)

There was a bug in CP 2.4.1, I just fixed it with CP 2.4.1a, you need to install an update with this method: https://vimeo.com/308801484
and everything should be working again.

**Comment by frankbash** [comment]
so i did it like in the video, but no chance to get it running,
when i put the commands via terminal all works fine, but not via control panel in chrome, there is not text feedback in the chrome window even too
so the commands seems correct, but the connection between cp and the rest is missing maybe
it´s like the cp is sending into the nirvana

**Comment by magdesign** [comment]
Now we are stuck in this issue: https://github.com/magdesign/PocketVJ-CP-v3/issues/38

If you Wifi connect to the PocketVJ, then opening Chrome and visit this link:
`http://2.0.0.100/backend.php?action=testscreen`
will then be the testscreen activate on the PocketVJ?

**Comment by frankbash** [comment]
chrome said http error 500, i change over to the issue #38

**Comment by ProjectileObjects** [comment]
Not sure if it will help, but I also recommend testing in a new private window.  Sometimes the browser stores the outdated information.

**Comment by frankbash** [comment]
nope, don´t help

**Comment by magdesign** [comment]
Could it be that you messed something while installing node.js and now the webserver is not running anymore?

**Comment by frankbash** [comment]
so i rolled back to the cp 2.4.0f (like in the alternative update video)and every works now again,
osc is running and cp via chrome browser

**Comment by magdesign** [comment]
Great.
